modules/social_features/social_event/modules/social_event_invite/src/SocialEventInviteAccessHelper.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Drupal\social_event_invite;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\node\NodeInterface;
+use Drupal\social_group\SocialGroupHelperService;
+
+/**
+ * Class SocialEventInviteAccessHelper
+ *
+ * @package Drupal\social_event_invite\Access
+ */
+class SocialEventInviteAccessHelper {
+
+  /**
+   * The route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * Configuration factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Group helper service.
+   *
+   * @var \Drupal\social_group\SocialGroupHelperService
+   */
+  protected $groupHelperService;
+
+  /**
+   * Entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * EventInvitesAccess constructor.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $routeMatch
+   *   The route match.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Configuration factory.
+   * @param \Drupal\social_group\SocialGroupHelperService $groupHelperService
+   *   The group helper service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\Core\Session\AccountProxyInterface $currentUser
+   *   The current user.
+   */
+  public function __construct(RouteMatchInterface $routeMatch, ConfigFactoryInterface $configFactory, SocialGroupHelperService $groupHelperService, EntityTypeManagerInterface $entityTypeManager, AccountProxyInterface $currentUser) {
+    $this->routeMatch = $routeMatch;
+    $this->configFactory = $configFactory;
+    $this->groupHelperService = $groupHelperService;
+    $this->entityTypeManager = $entityTypeManager;
+    $this->currentUser = $currentUser;
+  }
+
+  /**
+   * Custom access check for the event invite features for event managers.
+   *
+   * @return \Drupal\Core\Access\AccessResult
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function eventFeatureAccess() {
+    $config = $this->configFactory->get('social_event_invite.settings');
+    $enabled_global = $config->get('invite_enroll');
+
+    // If it's globally disabled, we don't want to show the block.
+    if (!$enabled_global) {
+      return AccessResult::forbidden();
+    }
+
+    // Get the group of this node.
+    $node = $this->routeMatch->getParameter('node');
+    if ($node instanceof NodeInterface) {
```

Comment:
The `$node` in this case is sometimes a full node object, sometimes just the ID.